base/error.jl

```diff
@@ -147,14 +147,15 @@ uncaught exceptions.
     This function went by the experimental name `catch_stack()` in Julia
     1.1–1.6, and had a plain Vector-of-tuples as a return type.
 """
-function current_exceptions(task::Task=current_task(); backtrace::Bool=true)
+function current_exceptions(task::Task=current_task(); backtrace::Bool=true, time::Bool=true)
     raw = ccall(:jl_get_excstack, Any, (Any,Cint,Cint), task, backtrace, typemax(Cint))::Vector{Any}
     formatted = Any[]
     stride = backtrace ? 3 : 1
     for i = reverse(1:stride:length(raw))
         exc = raw[i]
         bt = backtrace ? Base._reformat_bt(raw[i+1],raw[i+2]) : nothing
-        push!(formatted, (exception=exc,backtrace=bt))
+        tm = time ? ccall(:jl_clock_now, Any, ())::Any : nothing
+        push!(formatted, (exception=exc,backtrace=bt,time=tm))
```

Comment:
It might be beneficial to hide the type info for `exc` from the type, since it could be anything.
```suggestion
        NT = bt === nothing ? # manual union split, since inference currently can't
            @NamedTuple{exception::Any, backtrace::Nothing, time::typeof(tm)} : 
            @NamedTuple{exception::Any, backtrace::typeof(bt), time::typeof(tm)}
        push!(formatted, NT((exc, bt, tm)))
```

Reply:
Understood, I will do that by 7th (since its my birthday tomorrow)

Reply:
I tried this, it seems `NamedTuple` is not defined by the time this bit of code runs, I'm getting `UndefVarError` during build time. I'm leaving this bit as it was.